### ISSUE TITLE
[bug] java build macro Same Fieldname as Enumname

### ISSUE THREAD

**Issue by lublak** [HaxeFoundation/haxe#8056]
Error (only in java):

haxe 4.0 rc.2

`This expression cannot be generated because Test is shadowed by the current scope`

The Error shows on the enum Test; Test3(test:String); Test2 and Test.

Only throws if `enum Test` has the same name as one of his values.
It works if i remove the value Test.

Main.hx
```haxe
enum Test {
  Test3(test:String);
  Test2;
  Test;
}

@:build(Macro.build())
class Main {
  public var test:Test = Test2;
  private function new() {

  }
  static function main() {
    trace("Haxe is great!");
    new Main().testFN();
  }
}
```

Macro.hx
```haxe
class Macro {
  public static function build() {
    var fields = haxe.macro.Context.getBuildFields();
    var cls = haxe.macro.Context.getLocalClass().get();
    fields.push({
      name: 'testFN',
    	access: [APublic],
      kind: FFun({
        expr: macro {
          switch(test) {
            case Test3(test): trace(test);
            case Test:
            case Test2:
          }
        },
         ret: null,
        args: []
      }),
      pos: haxe.macro.Context.currentPos()
    });
    return fields;
  }
}
```

**Comment by kevinresol** [comment]
perhaps you can use `@:native('SomeOtherName') enum Test {` to workaround it

**Comment by lublak** [comment]
Thank you. I simple change the name of it. But i send it because it is (i think) a bug.
@kevinresol but here you get an heart: ❤️ 
Could it be a solution to prefix all "native" java enums? or is it gen bug?

**Comment by kevinresol** [comment]
I'd consider it as a generator bug. The generator is supposed to take care of native matters like this one.

P.S. this happens on C# as well, ref: https://github.com/haxetink/tink_core/blob/76a1e82db507edd79440b5dab91fe3bbd88c9e6c/src/tink/core/Noise.hx#L4

**Comment by Simn** [comment]
The Java and C# targets will be put in "regression fixes only" state with the Haxe 4.3.0 release. See https://github.com/HaxeFoundation/haxe/wiki/Target-Support-Level#java-and-c-since-430 for additional information. We apologize for any inconvenience this causes!
